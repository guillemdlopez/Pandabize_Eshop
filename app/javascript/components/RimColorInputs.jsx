import React from "react";
import PropTypes from "prop-types";
import { borderColor } from "../config/functions";

const RimColorInputs = ({
  rimColors,
  handleInputChange,
  setPrice,
  setPriceRim,
  priceRim,
  price,
  setErrorMsg,
  displayError,
  errorMsgContent,
}) => {
  return (
    <>
      <p className="text-muted text-center mb-0">Rim Color</p>
      <ul
        className="block-list-center customizations-list text-center animate__animated animate__fadeIn"
        id="custom-associations"
      >
        {rimColors.map((color) => {
          return (
            <li key={color.id} id="rim-colors" style={borderColor(color.value)}>
              <label htmlFor={color.id} className="mb-0 mr-2">
                {color.value}
              </label>
              <input
                type="radio"
                name="rimColor"
                key={color.id}
                value={color.id}
                onClick={(e) => {
                  handleInputChange(e);
                  setPriceRim((priceRim = 0));
                  setPriceRim(color.price);
                  setPrice(price + color.price);
                  setErrorMsg({ displayError: false, errorMsgContent: "" });
                }}
                data-id={color.id}
              />
            </li>
          );
        })}
      </ul>
    </>
  );
};

RimColorInputs.propTypes = {
  rimColors: PropTypes.array.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  setPrice: PropTypes.func.isRequired,
  setPriceRim: PropTypes.func.isRequired,
  price: PropTypes.number.isRequired,
  setErrorMsg: PropTypes.func.isRequired,
  displayError: PropTypes.bool.isRequired,
  errorMsgContent: PropTypes.string.isRequired,
};

export default RimColorInputs;
