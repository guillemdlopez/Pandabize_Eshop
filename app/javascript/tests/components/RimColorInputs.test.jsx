import { shallow } from "enzyme";
import React from "react";
import RimColorInputs from "../../components/RimColorInputs";

describe("Testing the <RimColorInputs /> component", () => {
  const rimColorInputsProps = {
    rimColors: [
      {
        id: 175,
        name: "Rim color",
        value: "Lightblue",
        price: 9.37,
        bicycle_id: 28,
      },
      {
        id: 176,
        name: "Rim color",
        value: "Orange",
        price: 5.52,
        bicycle_id: 28,
      },
      {
        id: 177,
        name: "Rim color",
        value: "Purple",
        price: 5.79,
        bicycle_id: 28,
      },
    ],
    handleInputChange: jest.fn(),
    setPrice: jest.fn(),
    setPriceRim: jest.fn(),
    price: 0,
    setErrorMsg: jest.fn(),
    displayError: false,
    errorMsgContent: "",
  };
  const wrapper = shallow(<RimColorInputs {...rimColorInputsProps} />);

  test("should render it correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should display a label and an input for each color", () => {
    const { rimColors } = rimColorInputsProps;

    const values = rimColors.map((color) => color.value);
    const [lightblue, orange, purple] = values;
    expect(wrapper.find("label").at(0).text()).toBe(lightblue);
    expect(wrapper.find("label").at(1).text()).toBe(orange);
    expect(wrapper.find("label").at(2).text()).toBe(purple);
    expect(wrapper.find("input").length).toBe(values.length);
  });
});
