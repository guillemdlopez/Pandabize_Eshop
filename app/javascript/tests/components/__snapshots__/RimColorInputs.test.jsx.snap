// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <RimColorInputs /> component should render it correctly 1`] = `
<Fragment>
  <p
    className="text-muted text-center mb-0"
  >
    Rim Color
  </p>
  <ul
    className="block-list-center customizations-list text-center animate__animated animate__fadeIn"
    id="custom-associations"
  >
    <li
      id="rim-colors"
      key="175"
      style={
        Object {
          "border": "2px solid rgb(173, 216, 230)",
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      <label
        className="mb-0 mr-2"
        htmlFor={175}
      >
        Lightblue
      </label>
      <input
        data-id={175}
        key="175"
        name="rimColor"
        onClick={[Function]}
        type="radio"
        value={175}
      />
    </li>
    <li
      id="rim-colors"
      key="176"
      style={
        Object {
          "border": "2px solid rgb(255, 165, 0)",
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      <label
        className="mb-0 mr-2"
        htmlFor={176}
      >
        Orange
      </label>
      <input
        data-id={176}
        key="176"
        name="rimColor"
        onClick={[Function]}
        type="radio"
        value={176}
      />
    </li>
    <li
      id="rim-colors"
      key="177"
      style={
        Object {
          "border": "2px solid rgb(128, 0, 128)",
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      <label
        className="mb-0 mr-2"
        htmlFor={177}
      >
        Purple
      </label>
      <input
        data-id={177}
        key="177"
        name="rimColor"
        onClick={[Function]}
        type="radio"
        value={177}
      />
    </li>
  </ul>
</Fragment>
`;
